### PR TITLE
Xref conceptual display property

### PR DESCRIPTION
**Pull request by live1206**
Support 
```
<xref:uid?displayProperty=fullName>
```

### CODE REVIEW

File: src/docfx/lib/markdown/ResolveXref.cs [hunk 1]
```diff
@@ -15,31 +16,49 @@ internal static class ResolveXref
...
-            });
+                         string display;
+                         var content = string.IsNullOrEmpty(xrefSpec.GetName()) ? xrefSpec.Uid : xrefSpec.GetName();
+                         if (!string.IsNullOrEmpty(query))
+                         {
+                             var queries = HttpUtility.ParseQueryString(query.Substring(1));
+                             var displayProperty = queries["displayProperty"];
+                             if (!string.IsNullOrEmpty(displayProperty))
+                             {
+                                 display = xrefSpec.GetXrefPropertyValue(displayProperty);
```
Comment (yufeih) [inline src/docfx/lib/markdown/ResolveXref.cs]: display could be null if the property does not exists, suggest to use the following fallback:

```csharp

propertyName = query?.displayName ?? "name"
display = xrefspec.ExtensionData[propertyName] ?? uid

```

File: docs/specs/xref.yml [hunk 2]
```diff
@@ -93,6 +93,44 @@ outputs:
...
+      "name": "test",
+      "fullName": "test.test",
+      "description": "a.description"
+    }
+  docs/c.md: Link to <xref:a?displayProperty=fullName>
+outputs:
+  docs/c.json: | 
+    {"content":"<p>Link to <a href=\"docs/a.json\">test.test</a></p>\n"}
+  docs/a.json:
+  xrefmap.json: |
```
Comment (yufeih) [inline docs/specs/xref.yml]: maybe merge these 2 test cases to save some test time

File: src/docfx/build/xref/XRefSpec.cs [hunk 3]
```diff
@@ -15,6 +15,9 @@ public class XrefSpec
         [JsonExtensionData]
         public JObject ExtensionData { get; } = new JObject();
 
-        public string GetName() => ExtensionData.TryGetValue("name", out var name) && name is JValue v && v.Value is string str ? str : null;
+        public string GetName() => GetXrefPropertyValue("name");
+
+        public string GetXrefPropertyValue(string property)
+            => ExtensionData.TryGetValue(property, out var prop) && prop is JValue v && v.Value is string str ? str : null;
```
Comment (OsmondJiang) [inline src/docfx/build/xref/XRefSpec.cs]: >ExtensionData.TryGetValue(property, out var prop) && prop is JValue  [](start = 15, length = 68)

we have a JosnUtility.TryGetValue<>();

Comment (live1206) [reply src/docfx/build/xref/XRefSpec.cs]: ok, good to know

**Pull request merged**